backend/uchicago/src/coursesearch_parser.py

```diff
@@ -0,0 +1,41 @@
+import collections
+import re
+import scrapydo
+
+from .course import Course
+from .section import Section
+
+from .coursesearch_spider import CourseSearchSpider
+
+class CourseSearch(object):
+  init = False
+  def __init__(self, label, id):
+    if not CourseSearch.init:
+      scrapydo.setup()
+      CourseSearch.init = True
+    self.label = label
+    self.id = id
+    
+  @property
+  def courses(self):
+  
+    settings = {
+        'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)',
+        'DOWNLOADER_MIDDLEWARES': {
+            'src.coursesearch_spider.StateManagerMiddleware': 543,
+        }
+    }
+    
+    spider = CourseSearchSpider
+    ret = scrapydo.run_spider(spider, settings=settings, label = self.label, id = self.id)
```

Comment:
The math department seems to float around 150 courses.  See the counts I generated here: http://www.sharecsv.com/s/80aba80aea72ca38df2e0f3ee0c01012/winter-course-count.csv

The issue with day of week sharding is that it will miss classes that do not have a timeslot assigned.  For example, EVOL, which is the only one winter quarter of this year that hits the cap, is mostly full of timeslot-free reading/research courses.